net/private_net.go

```diff
@@ -41,3 +44,15 @@ func IsPrivateAddress(ip net.IP) bool {
 	}
 	return false
 }
+
+// RestrictIP returns an error if the destination IP is not a
+// standard public IP.
+func RestrictIP(ip net.IP) *ConnectionError {
```

Comment:
Please rename so it reflects the filtering.  Maybe something like `ensurePublicIP()`, `verifyPublicIP`, `assertPublicIP`?

Reply:
Renamed to "RequirePublicIP".